app/models/document.rb

```diff
@@ -104,12 +104,26 @@ def humanized_attributes
     end
   end
 
+  def self.extract_body_from_payload(payload)
+    body_attribute = payload['details']['body']
+
+    case
+    when body_attribute.class == Array
+      govspeak_body = body_attribute.detect do |body_hash|
+        body_hash['content_type'] == 'text/govspeak'
+      end
+      govspeak_body['content']
+    when body_attribute.class == String
+      body_attribute
+    end
+  end
```

Comment:
Should this be more like:

```ruby
case body_attribute
when Array
  # ...
when String
  # ...
end
```

Case-when statements test conditionals with the triple equals (===) operator, which tests the class:
http://ruby-doc.org/core-2.3.0/Object.html#method-i-3D-3D-3D